src/main/java/Duke.java

```diff
@@ -1,59 +1,122 @@
-        if (num >= allTasks.size()) {
+    // print newly added task
+    public static void printNewlyAddedTask(Task task) {
+        System.out.println(indent + "Got it. I've added this task: "
+                + System.lineSeparator()
+                + indent + indent + task
+                + System.lineSeparator()
+                + indent + "Now you have " + allTasks.size() + " tasks in the list.");
+    }
+
+    // mark a Task as done
+    public static void markTaskAsDone(int num) {
+        if (num > allTasks.size() || num == 0) {
             System.out.println(indent + "Sorry, this task does not exist!");
         }
         else {
-            allTasks.get(num - 1).setAsDone();
+            allTasks.get(num - 1).setIsDone(true);
             System.out.println(indent + "Nice! I've marked this task as done:");
             System.out.println(indent + "[\u2713] " + allTasks.get(num - 1).getTask());
         }
     }
 
-    public static void addTask(String input) {
-        if (!input.isEmpty()) {
-            Task newTask = new Task(input);
-            allTasks.add(newTask);
-            System.out.println(indent + "added: " + input);
-        }
-        else {
-            System.out.println(indent + "No text specified.");
+    public static void addTask(String[] inputArr) {
+
+        /* TODO: implement error checking for user input
+           cases:
+           - input is completely empty
+           - inputArr[0] == _todo/deadline/event, but rest of the input is completely empty
+           - deadline/event does not have "/by" or "/at"
+        */
+
+        Task newObj;
+
+        switch(inputArr[0]) {
+            case("todo"): {
+                String[] descArr = Arrays.copyOfRange(inputArr, 1, inputArr.length);
+                newObj = new Todo((convertToString(descArr)));
+                break;
+            }
+            case("deadline"): {
+                int pos = Arrays.asList(inputArr).indexOf("/by");
+                String[] descArr = Arrays.copyOfRange(inputArr, 1, pos);
+                String[] byArr = Arrays.copyOfRange(inputArr,pos+1,inputArr.length);
+                newObj = new Deadline(convertToString(descArr),convertToString(byArr));
+                break;
+            }
+            case("event"): {
+                int pos = Arrays.asList(inputArr).indexOf("/at");
+                String[] descArr = Arrays.copyOfRange(inputArr, 1, pos);
+                String[] atArr = Arrays.copyOfRange(inputArr,pos+1,inputArr.length);
+                newObj = new Event(convertToString(descArr),convertToString(atArr));
+                break;
+            }
+            default: {
+                newObj = new Task((convertToString(inputArr)));
+            }
         }
+        allTasks.add(newObj);
+        printNewlyAddedTask(newObj);
     }
 
+
     public static void main(String[] args) {
 
         System.out.println(indent + "Hello! I'm Duke\n" + indent + "What can I do for you?\n");
```

Comment:
To add on to what Zhi-You have pointed out, perhaps this line could be refactored out as a method? This would help the code adhere closer to SLAP